R/callback-validation-check.R

```diff
@@ -0,0 +1,74 @@
+#' @include callbacks.R
+NULL
+
+#' Validation Check
+#'
+#' Check validation loop before fitting model.
+#'
+#' @param batches Number of validation batches to check. Default is 2.
+#'
+#' @note Usually the training loop is much longer than the validation
+#'   loop and issues with the validation loop aren't encountered until after
+#'   a long training runtime. This callback runs the validation loop first on
+#'   `batches` number of batches and then proceeds onto the standard
+#'   training process.
+#'
+#' @note Printing can be disabled by passing `verbose = FALSE` to
+#'   [fit.luz_module_generator()].
+#'
+#' @family luz_callbacks
+#'
+#' @returns
+#' A `luz_callback`.
+#'
+#' @export
+luz_callback_validation_check <- luz_callback(
+  "validation_check_callback",
+  initialize = function(batches = 2) {
+    if (!rlang::is_scalar_integerish(batches)) {
+      rlang::abort("`batches` must be a single integer value.")
+    }
+    self$batches <- batches
+  },
+  on_fit_begin = function() {
+    if (is.null(ctx$valid_data)) return()
+    if (self$batches <= 0) return()
+
+    ctx$model$eval()
```

Comment:
We might want to extract this out into a function:

https://github.com/mlverse/luz/blob/6e0bb77f61e8f2f58f3f177c90d1e2af4ec579bc/R/callbacks.R#L298-L300

And reuse here so we make sure that the same changes are always set?